src/Goteo/Repository/DataSetRepository.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/*
+ * This file is part of the Goteo Package.
+ *
+ * (c) Platoniq y Fundación Goteo <fundacion@goteo.org>
+ *
+ * For the full copyright and license information, please view the README.md
+ * and LICENSE files that was distributed with
+ */
+
+ namespace Goteo\Repository;
+
+ use Goteo\Application\Exception\ModelException;
+ use Goteo\Application\Exception\ModelNotFoundException;
+ use Goteo\Model\DataSet;
+ use PDO;
+ use PDOException;
+
+ class DataSetRepository extends BaseRepository {
+     protected ?string $Table = 'data_set';
+
+     public function getById(int $id): DataSet  {
+         $sql = "SELECT data_set.*
+                FROM data_set
+                WHERE data_set.id = ?";
+
+         $dataSet = $this->query($sql, [$id])->fetchObject(DataSet::class);
+
+         if (!$dataSet instanceOf DataSet)
+             throw new ModelNotFoundException("DataSet with id $id not found");
+
+         return $dataSet;
+    }
+
+    public function getList(array $filter, int $offset = 0, int $limit = 10, bool $count = false) {
+         if ($count) {
+             $sql = "SELECT count(data_set.*)
+                FROM data_set
+                LIMIT $limit
+                OFFSET $offset";
+
+             return $this->query($sql)->fetchAll();
+         }
+
+         $sql = "SELECT data_set.*
+                FROM data_set
+                LIMIT $limit
+                OFFSET $offset";
+
+        return $this->query($sql)->fetchAll(PDO::FETCH_CLASS, DataSet::class );
+    }
+
+    public function getListByFootprint(array $footprints) {
```

Comment:
If `$footprints` is empty, I'm pretty confident that the SQL doesn't need even to be executed. So returning an empty list will most likely be more efficient. 

And in this case, adding the contents of `$sqlWhere` directly into the `$sql` will be more concise.

Or am I missing something @davidbeig ?